src/arithmetic.jl

```diff
@@ -54,13 +54,56 @@ for TM in tupleTMs
 
         *(b::T, a::$TM) where {T<:NumberNotSeries} = $TM(a.pol*b, b*a.rem, a.x0, a.dom)
 
+        # Multiplication
+        function *(a::$TM, b::$TM)
+            @assert tmdata(a) == tmdata(b)
+            a_order = get_order(a)
+            b_order = get_order(b)
+            rnegl_order = a_order + b_order
+            aux = a.dom - a.x0
+
+            # Returned polynomial
+            res = a.pol * b.pol
+            order = res.order
+
+            # Remainder of the product
+            if $TM == TaylorModel1
+
+                # Remaing terms of the product as reduced Taylor1 (factored polynomial)
+                rnegl = Taylor1(zero(res[0]), rnegl_order)
+                for k in order+1:rnegl_order
+                    @inbounds for i = 0:k
+                        (i > a_order || k-i > b_order) && continue
+                        rnegl[k] += a.pol[i] * b.pol[k-i]
+                    end
+                end
+
+                # Bound for the neglected part of the product of polynomials
+                Δnegl = rnegl(aux)
```

Comment:
Line 73 effectively creates a `Taylor1` polynomial of the needed order to compute the remaining terms of the polynomial product. This is ok, but the order of `Taylor1` is `a.order+b.order` (so we use memory to store trivial zeros). I wonder if we can improve this, by constructing only a polynomial of order `rnegl_order - order`.

The idea is to mimic factorizing t^(order+1) from `rnegl`, create the resulting polynomial, and then bound that product with the usual interval methods. I've played with this idea but some of the tests fail, since itleads to wider intervals, which I guess is due to the dependency problem. 

@dpsanders Do you have any idea/suggestion how to get around this?